src/components/SavedSearchRenderer.js

```diff
@@ -0,0 +1,253 @@
+  location: PropTypes.object.isRequired,
+};
+
+type SavedSearchRendererDefaultProps = {};
+
+type SavedSearchRendererState = {
+  response?: QueryResponse,
+  error?: string,
+  showSaveSearchModal: boolean,
+  savedSearchTitle: string,
+  savedSearchList: Array<SavedSearch>,
+};
+
+/**
+ * Component to include in the Masthead for entering the query
+ * to use when searching. Must be inside a Searcher component.
+ */
+class SavedSearchRenderer extends React.Component<
+  SavedSearchRendererDefaultProps,
+  SavedSearchRendererProps,
+  SavedSearchRendererState,
+> {
+  static defaultProps: SavedSearchRendererDefaultProps = {};
+
+  static displayName = 'SavedSearchRenderer';
+
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  constructor(props: SavedSearchRendererProps) {
+    super(props);
+    this.state = {
+      response: undefined,
+      error: undefined,
+      showSaveSearchModal: false,
+      savedSearchTitle: '',
+      savedSearchList: [],
+    };
+    (this: any).applySavedSearch = this.applySavedSearch.bind(this);
+    (this: any).showSaveSearchModal = this.showSaveSearchModal.bind(this);
+    (this: any).hideSaveSearchModal = this.hideSaveSearchModal.bind(this);
+    (this: any).searchTitleEntered = this.searchTitleEntered.bind(this);
+    (this: any).renderSavedQuery = this.renderSavedQuery.bind(this);
+    (this: any).renderSavedQueryHeader = this.renderSavedQueryHeader.bind(this);
+  }
+
+  state: SavedSearchRendererState;
+
+  componentWillMount() {
+    this.context.metadataManager.getSavedSearches();
+    this.setState({
+      savedSearchList: this.context.metadataManager.state.savedSearchList,
+    });
+  }
+
+  // applies the saved search which includes a query and facet filters
+  applySavedSearch(query: string) {
+    const { location } = this.props;
+    this.props.history.push({
+      pathname: location.pathname,
+      search: query,
+    });
+  }
+
+  searchTitleEntered(e: Event) {
+    if (e.target instanceof HTMLInputElement) {
+      this.setState({
+        savedSearchTitle: e.target.value,
+      });
+    }
+  }
+
+  // hides the modal
+  hideSaveSearchModal() {
+    this.setState({ showSaveSearchModal: false });
+  }
+
+  // sets the query from searcher as the search title and shows the modal
+  showSaveSearchModal() {
+    this.setState({
+      savedSearchTitle: this.context.searcher.state.query,
+      showSaveSearchModal: true,
+    });
+  }
+
+  renderSavedQueryHeader() {
+    return (
+      <MenuItem
+        key="saveQueryHeader"
+        onSelect={() => {
+          this.showSaveSearchModal();
+        }}
+        id="saveQueryHeader"
+      >
+        <span
+          style={{
+            fontSize: '14px',
+            color: '#003c7e',
+          }}
+        >
+          <b> SAVE SEARCH </b>
+        </span>
+      </MenuItem>
+    );
+  }
+
+  renderSavedQueryList() {
+    return this.state.savedSearchList.length > 0 ? (
+      this.state.savedSearchList.map((e) => {
```

Comment:
Since this is just a return with any other declarations, you can use the auto-return feature of arrow functions. Just use parentheses with no curly braces and leave out the return. If you are using auto return and only have one param, that param does not have to be in parentheses either.
```
this.state.savedSearchList.map(e => (
  <MenuItem key={e.id}>
    ...
  <MenuItem />
);
```